terminal-emulator/src/test/java/com/termux/terminal/TerminalTest.java

```diff
@@ -212,21 +217,21 @@ public void testSelectGraphics() {
 		enterString(("\033[0m")); // Reset fg and bg colors.
 		assertEquals(TextStyle.COLOR_INDEX_FOREGROUND, mTerminal.mForeColor);
 		assertEquals(TextStyle.COLOR_INDEX_BACKGROUND, mTerminal.mBackColor);
-		enterString("\033[38;2;255;127;2;48;2;1;2;254m");
+		enterString("\033[38;2;255;127;2;48;2;1;2;254m".replace(';', separator));
```

Comment:
You don't want to replace all semicolons with colons. Having all separators be colon would in all other terminal emulators (at least the 6 I tested) only set the foreground color. Instead you want the separator between the colors to remain a semicolon. So these are the two cases you want to test, which should both render as orange text on blue background:

```
\033[38;2;255;127;2;48;2;1;2;254m
\033[38:2:255:127:2;48:2:1:2:254m
```

I haven't tested this code, but if the one with all colons (`\033[38:2:255:127:2:48:2:1:2:254m`) renders the same as the one with semicolons here, note that this is different behavior from other terminal emulators.

Reply:
@trygveaa Thanks, you are right! I just updated the behaviour now so that extra SGR sub-parameters are skipped.

Please add any further review or testing notes, really appreciated :bow:.